src/nfa/nfa.cc

```diff
@@ -1546,17 +1363,18 @@ Nfa::const_iterator Nfa::const_iterator::for_begin(const Nfa* nfa)
     assert(nullptr != nfa);
 
     const_iterator result;
-    if (nfa->nothing_in_trans())
+    if (std::all_of(nfa->transition_relation.begin(), nfa->transition_relation.end(),
```

Comment:
Hm, this can be quite costly. Maybe we could optimize in future, e.g. by having some boolean flag `has_no_transitions`, that is initialized to `true` and whenever we add transition we set `self.has_no_transitions = false`? Could be a little bit more efficient.

Reply:
generally I don't like adding variables making code more state dependant. I think it is should not be so costly compared to time we may spend debugging statefull code.

Reply:
Let it leave as it is for now. If it show up as a performance bottleneck we could refactore it later.